_posts/2022-XX-XX-fido2-central-auth.md

```diff
@@ -0,0 +1,130 @@
+---
+layout:     post
+title:      "Local authentication in a centralized management system with FIDO2"
+date:       2022-09-08 12:00:00 +0100
+categories: idm
+---
+
+#TODO: change date
+
+#TODO: change for fido2 helper process for sssctl
+
+# Introduction
+
+I have been working on a feature that will allow centrally managed users to log in locally to a computer with FIDO2 keys. As the feature is still under development I've prepared an environment to test it.
+
+## Dependencies
+
+The test environment uses docker/podman to set up the containers and ansible to provision them. On top of that, yubikey-manager is needed to manage the FIDO2 device options. Moreover, I've created a [COPR repository](https://copr.fedorainfracloud.org/coprs/ipedrosa/fido2-central-auth/) to deliver the patched versions of SSSD and IPA that will enable this feature. Finally, I've prepared a new [sssd-ci-container branch](https://github.com/ikerexxe/sssd-ci-containers/tree/fido2) to automate the provision of everything. The following sections will explain how to set up the environment and how to test it.
+
+# How to set up the environment
+
+1. Clone sssd-ci-containers locally:
+
+```console
+git clone https://github.com/ikerexxe/sssd-ci-containers/
+git checkout --track origin/fido2
+```
+
+2. Host package installation and configuration. The following commands should be executed only once:
+
+```console
+$ sudo dnf install -y podman podman-docker docker-compose yubikey-manager
+$ sudo systemctl enable --now podman.socket
+$ sudo setsebool -P container_manage_cgroup true
+$ cp env.example .env
+```
+
+3. Open the cloned sssd-ci-containers project:
+
+```console
+cd sssd-ci-containers
+```
+
+4. Set up the containers (make up) and update the packages with the FIDO2 patches (make fido2). For the latter command to run correctly you will need to input the password `Secret123`. It will take some time to execute as it needs to update the dnf cache and then update the sssd packages.
+
+```console
+$ sudo make up

```

Comment:
I got the following error, which I don't see with upstream sssd-ci-containers

~~~
[justin@agalloch sssd-ci-containers]$ sudo make up
docker-compose up --detach 
Recreating ipa ... 
Recreating dns    ... done
Recreating ipa    ... done
Recreating client ... error
Recreating ldap   ... done

Recreating samba  ... done

ERROR: for client  Cannot create container for service client: container create: invalid config provided: CapAdd and privileged are mutually exclusive options
ERROR: Encountered errors while bringing up the project.
make: *** [Makefile:10: up] Error 1
~~~